tests/integration/test_charm.py

```diff
@@ -2,28 +2,46 @@
 # Copyright 2022 Canonical Ltd.
 # See LICENSE file for licensing details.
 
-
 import logging
-import urllib.request
-from pathlib import Path
 
 import pytest
-import yaml
 from pytest_operator.plugin import OpsTest
 
-logger = logging.getLogger(__name__)
+from tests.integration.helpers import APP_NAME, SERIES, UNIT_IDS
 
-METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
-APP_NAME = METADATA["name"]
+logger = logging.getLogger(__name__)
 
 
+@pytest.mark.charm_tests
 @pytest.mark.abort_on_fail
-async def test_application_is_up(ops_test: OpsTest):
-    status = await ops_test.model.get_status()  # noqa: F821
-    address = status["applications"][APP_NAME]["units"][f"{APP_NAME}/0"]["address"]
-
-    url = f"http://{address}"
-
-    logger.info("querying app address: %s", url)
-    response = urllib.request.urlopen(url, data=None, timeout=2.0)
-    assert response.code == 200
+@pytest.mark.skip_if_deployed
+async def test_build_and_deploy(ops_test: OpsTest) -> None:
+    """Build and deploy one unit of MongoDB."""
+    my_charm = await ops_test.build_charm(".")
+    await ops_test.model.set_config(
+        {
+            "logging-config": "<root>=INFO;unit=DEBUG",
+            "update-status-hook-interval": "1m",
+            "cloudinit-userdata": """postruncmd:
+                - [ 'sysctl', '-w', 'vm.max_map_count=262144' ]
+                - [ 'sysctl', '-w', 'fs.file-max=1048576' ]
+                - [ 'sysctl', '-w', 'vm.swappiness=0' ]
+                - [ 'sysctl', '-w', 'net.ipv4.tcp_retries2=5' ]
+        """,
+        }
+    )
+
+    await ops_test.model.deploy(
+        my_charm,
+        num_units=len(UNIT_IDS),
+        series=SERIES,
+    )
+    await ops_test.model.wait_for_idle()
+
+
+@pytest.mark.charm_tests
+@pytest.mark.abort_on_fail
+async def test_status(ops_test: OpsTest) -> None:
+    """Verifies that the application and unit are active."""
+    await ops_test.model.wait_for_idle(apps=[APP_NAME], status="blocked", timeout=1000)
```

Comment:
I'm waiting for Blocked, because without having the TLS relation established, OpenSearch should be in a BlockedState "Waiting for TLS to be configured..." as you can see [here](https://github.com/canonical/opensearch-operator/blob/fixes-integ_tests-init-charm-tls/src/charm.py#L101-L102) - Unless I missed something, it doesn't seem like there is a way to wait for a certain status + certain message

Reply:
No you're right, just confirming that it's how it's meant to be working :)